numpyro/distributions/flows.py

```diff
@@ -78,6 +89,9 @@ def log_abs_det_jacobian(self, x, y):
         :param y: the output of the transform
         :type y: numpy array
         """
-        log_scale = self.arn.apply_fun(self.params, x)[..., 1, :]  # TODO: this shouldn't be recomputed here
-        log_scale = _clamp_preserve_gradients(log_scale, self.log_scale_min_clip, self.log_scale_max_clip)
+        if self.caching:
+            log_scale = self._cached_log_scale
```

Comment:
Okay, I see the issue that you mentioned. Yes, we'll need to pass in `intermediates` to `log_abs_det_jacobian`. Basically, `TransformedDistribution.sample_with_intermediates` returns a list of `(x, transform_intermediates)` and `transform_intermediates` is passed down to `log_abs_det_jacobian`. We don't need to enforce what `transform_intermediates` is - it could be an array, list or None (by default). WDYT?

Reply:
That makes sense to me. I'll follow that and make a separate PR which can resolve caches for log_det method.

Reply:
The changes look good to me otherwise! I'll take another pass once this is resolved.

Reply:
@neerajprad I just remove caching stuff and incorporate intermediates stuff. For a test of IAF guide, I'll do it at #224. The changes here only exercise the test `test_transformed_distribution_intermediates`, which confirms the adding of `call_with_intermediates/log_abs_det_jacobian_with_intermediates` method do not affect current behaviours of transformed distributions.

Reply:
Great, taking a look at this.